containers/unit_tests/TestVector.hpp

```diff
@@ -298,6 +298,36 @@ TEST(TEST_CATEGORY, vector_insert) {
   Impl::test_vector_insert<int, TEST_EXECSPACE>(3057);
 }
 
+TEST(TEST_CATEGORY, vector_push_back_default_exec) {
```

Comment:
Pick a better name or comment on why this test was added and what bug it exposes.  Optionally add a link to the issue.